sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -81,6 +89,141 @@ def __init__(
             )
         self.job_kind = job_kind
 
+    def train(
+        self,
+        name: str = None,
+        namespace: str = None,
+        num_workers: int = 1,
+        num_procs_per_worker: int = 1,
+        storage_config: Dict[Literal["size", "storage_class"], str] = None,
+        model_provider_parameters: HuggingFaceModelParams = None,
+        dataset_provider_parameters: S3DatasetParams = None,
+        train_parameters: HuggingFaceTrainParams = None,
+        resources_per_worker: Dict[Literal["gpu", "cpu", "memory"], any] = None,
+    ):
+        """
+        Higher level train api
+        """
+        if (
+            not name
+            or not namespace
+            or not storage_config
+            or not model_provider_parameters
+            or not dataset_provider_parameters
+            or not train_parameters
+            or not resources_per_worker
+        ):
+            raise ValueError("One of the required parameters is None")
+
+        try:
+            self.core_api.create_namespace(
+                body=utils.get_namespace_spec(namespace=namespace)
+            )
+        except Exception as e:
+            print(e)
+
+        PVC_NAME = "train-job-pvc"
+        self.core_api.create_namespaced_persistent_volume_claim(
+            namespace=namespace,
+            body=utils.get_pvc_spec(
+                pvc_name=PVC_NAME,
+                namespace=namespace,
+                storage_size=storage_config["size"],
+                storage_class=storage_config["storage_class"],
+            ),
+        )
```

Comment:
That will require changes in Training Operator user RBAC: https://github.com/kubeflow/training-operator/blob/master/manifests/overlays/kubeflow/kubeflow-training-roles.yaml

Reply:
@andreyvelich why is this change needed? I was able to create the pvc without changing any permissions.

Reply:
@deepanker13 I think, you were able to create PVC since you have admin access to your Cluster.
When you use Kubeflow Profiles to create user roles, user has the following permission: https://github.com/kubeflow/training-operator/blob/master/manifests/overlays/kubeflow/kubeflow-training-roles.yaml#L17.
So when you use this API with that role, use has to have access to create PVCs.